### DISCUSSION TITLE
[Closed] Rename `EqualsConverter` + `NotEqualsConverter` to `IsEqualConverter` + `IsNotEqualConverter`

### DISCUSSION THREAD

**Discussion by TheCodeTraveler** [CommunityToolkit/Maui#314] [General]
Currently `EqualsConverter` and `NotEqualsConverter` are the only converters in `CommunityToolkit.Maui` that return a `bool` and do **not** begin with `Is*`.

`CommunityToolkit.Maui` includes the following converters which all return a `bool` and begin with `Is*`:
- `IsListNotNullOrEmptyConverter`
- `IsListNullOrEmptyConverter`
- `IsStringNotNullOrEmptyConverter`
- `IsStringNotNullOrWhiteSpaceConverter`
- `IsStringNullOrEmptyConverter`
- `IsStringNullOrWhiteSpaceConverter`

I propose that we update the following converters to match the existing naming schema and improve discoverability:
- Rename `EqualsConverter` to `IsEqualConverter`
- Rename `NotEqualsConverter` to `IsNotEqualConverter`

**Comment by VladislavAntonyuk** [comment]
as for the file name. some of them end with shared. some of them not:
should we also be consistent with the filename?
![image](https://user-images.githubusercontent.com/33021114/156830861-5ada674f-0207-4a15-994b-d668f173251f.png)

**Reply by TheCodeTraveler** [reply]
Also a good catch! We should fix that too 💯 

I wonder if @pictos knows of a way to have an analyzer create a compiler error any time a file is missing a suffix? 

E.g. Valid: `*.shared.cs`, `*.macios.cs`, `*.android.cs` etc
E.g. Invalid: `*.cs`

**Reply by VladislavAntonyuk** [reply]
https://github.com/CommunityToolkit/Maui/pull/315

**Reply by pictos** [reply]
I could take a look, but I'm 80% sure that Roslyn analyzers can verify the file name

**Reply by TheCodeTraveler** [reply]
Nice! I opened Discussion https://github.com/CommunityToolkit/Maui/discussions/316 for the file name validation 🙌

**Comment by CliffAgius** [comment]
Closed as answered
